include/bulk/world.hpp

```diff
@@ -81,12 +99,27 @@ class world {
   /**
    * Perform a global barrier synchronization of the active processors and
    * resolves any outstanding communication. Messages previously received in
-   * queues are cleared for the next superstep, unless `clear_queues` is set
-   * to false. The function must be called by all processors. When some
+   * queues are cleared for the next superstep, 
+   * unless the member `clear_queues` of sync_option is set
+   * to false. One may provide a tag to label the current call.
+   * The function must be called by all processors. When some
    * processors call `sync` while others call `barrier` at the same time,
    * behaviour is undefined.
    */
-  virtual void sync(bool clear_queues = true) = 0;
+  virtual void sync(sync_options option={}) = 0;
+
```

Comment:
```suggestion
void sync(bool clear_queues) { sync({.clear_queues = clear_queues}); }

```

to maintain backwards compatibility